crates/core/src/workflows/get_link_result.rs

```diff
@@ -34,20 +39,101 @@ pub async fn get_link_result_workflow<'a>(
 
     match links_result {
         GetLinksNetworkResult::Links(links) => {
-            let get_links_result = links
+            links
                 .into_iter()
-                .map(|get_entry_crud| LinksResult {
-                    address: get_entry_crud.target.clone(),
-                    headers: get_entry_crud.headers.unwrap_or_default(),
-                    status: get_entry_crud.crud_status,
-                    tag: get_entry_crud.tag.clone(),
+                .map(|(link_add_address, tag)| {
+                    // make DHT calls to get the entries for the links
+                    get_link_data_from_link_addresses(
+                        context,
+                        &link_add_address,
+                        &tag,
+                        link_args.options.headers,
+                    )
                 })
-                .collect::<Vec<LinksResult>>();
-
-            Ok(GetLinksResult::new(get_links_result))
+                .map(|maybe_get_entry_crud| {
+                    maybe_get_entry_crud.map(|get_entry_crud| LinksResult {
+                        address: get_entry_crud.target.clone(),
+                        headers: get_entry_crud.headers.unwrap_or_default(),
+                        status: get_entry_crud.crud_status,
+                        tag: get_entry_crud.tag.clone(),
+                    })
+                })
+                .collect::<Result<Vec<LinksResult>, HolochainError>>()
```

Comment:
Cool rust trick! If you collect an iterator of `Results` into a `Result<Vec<_>>` it will abort on the first `Err` and return that and only return `Ok` if none of the elements are `Err`

Reply:
This is where we can also decide how we want to handle failures to retrieve `LinkAdd` entries. Options are:

- Fail if any of them cannot be retrieved (current implementation, fits best with existing code)
- Filter out failures and only return successes
- get_links returns a vec of results (or possible separate vecs for errors and results)